pkg/sql/pgwire/hba/conf.rl

```diff
@@ -53,13 +55,17 @@ func Parse(input string) (*Conf, error) {
 		action mark { mark = p }
 
 		action quotedString {
-			s = String{
-				Value: string(data[mark:p-1]),
-				Quoted: true,
-			}
+			all = false
+			s = tree.Name(data[mark:p-1])
 		}
 		action string {
-			s = String{Value: string(data[mark:p])}
+			s = tree.Name(string(data[mark:p]))
+			all = false
+			if s == "all" {
```

Comment:
I'm opposed to hard-coding special keywords in the parser because, although we only support `all` now, we could easily support others in the future. https://www.postgresql.org/docs/current/auth-pg-hba-conf.html lists a bunch they support that we could easily want to use like `sameuser` and `samerole`, or come up with some on our own.